src/contexts/ConfigurationContext.tsx

```diff
@@ -409,7 +409,7 @@ export const ConfigContextProvider = ({ children }: Props) => {
 
     };
     loadConfig();
-  }, []);
+  }, [rerank.numResults]);
```

Comment:
I suspect this is going to trigger a redundant execution of this effect, because `rerank` is defined on line 407 when we call `setRerank`. Have you tested to see if that happens?